source/dofs/dof_tools_constraints.cc

```diff
@@ -1083,6 +1085,77 @@ namespace DoFTools
         }
     }
 
+    namespace internal
+    {
+      /**
+       * Try to find a finite element inside FECollection of @p dof_handler
+       * which is dominating those in @fes .
+       */
+      template <int dim, int spacedim>
+      unsigned int
+      find_dominating_fe (const dealii::hp::DoFHandler<dim,spacedim> &dof_handler,
+                          const std::set<unsigned int> fes)
+      {
+        unsigned int fe = numbers::invalid_unsigned_int;
+        unsigned int n_dofs = 0;
+        for (unsigned int cur_fe = 0; cur_fe < dof_handler.get_fe().size(); cur_fe++)
+          if (fes.find(cur_fe) == fes.end())
+            {
+              FiniteElementDomination::Domination domination = FiniteElementDomination::no_requirements;
+              // check if cur_fe can dominate all FEs in @p fes:
+              for (std::set<unsigned int>::const_iterator it = fes.cbegin();
+                   it!=fes.cend(); ++it)
+                {
+                  Assert (*it < dof_handler.get_fe().size(),
+                          ExcInternalError());
+                  domination = domination &
+                               dof_handler.get_fe()[cur_fe].compare_for_face_domination
+                               (dof_handler.get_fe()[*it]);
+                }
+
+              // if we found dominating element, compare it's number of dofs to
+              // those we have as our best bet. Choose the one with higher number
+              // of dofs.
+              // This is quick way to find the largest FE space among
+              // those present in FECollection.
+              // TODO: implement some function in FE class to decide which
+              // space includes which, i.e. Q(2) \in Q(3).
+              const unsigned int cur_dofs =  dof_handler.get_fe()[cur_fe].n_dofs_per_cell();
+              if (domination == FiniteElementDomination::this_element_dominates &&
+                  cur_dofs > n_dofs )
+                {
+                  n_dofs = cur_dofs;
```

Comment:
actually i do. It's a dirty trick to find the largest FEs dominating given FEs, i.e. the least dominant.